core-java-8/src/test/java/com/baeldung/encoderdecoder/EncoderDecoder.java

```diff
@@ -8,31 +8,87 @@
 import java.net.URLDecoder;
 import java.net.URLEncoder;
 import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
 
 public class EncoderDecoder {
 
     @Test
-    public void givenPlainURL_whenUsingUTF8EncodingScheme_thenEncodeURL() throws UnsupportedEncodingException {
-        String plainURL = "http://www.baeldung.com" ;
-        String encodedURL = URLEncoder.encode(plainURL, StandardCharsets.UTF_8.toString());
+    public void givenRequestParam_whenUTF8Scheme_thenEncode(){
+        Map<String, String> requestParams = new HashMap<>();
+        requestParams.put("key1", "value 1");
+        requestParams.put("key2", "value@!$2");
+        requestParams.put("key3", "value%3");
 
-        Assert.assertThat("http%3A%2F%2Fwww.baeldung.com", CoreMatchers.is(encodedURL));
+        StringBuilder requestParamsString = new StringBuilder();
+        requestParams.forEach((key, value) -> {
+            try {
+                requestParamsString
+                        .append(key)
+                        .append("=")
+                        .append(URLEncoder.encode(value,
+                                StandardCharsets.UTF_8.toString()))
+                        .append("&");
+            } catch (UnsupportedEncodingException e) {
```

Comment:
Let's not try-catch exceptions in tests. We can simply add this to a method signature just like it was before